src/qgitrepository.cpp

```diff
@@ -624,4 +627,17 @@ bool Repository::shouldIgnore(const QString &path) const
     return result;
 }
 
+void Repository::setIdent(const QString &name, const QString &email)
+{
+    qGitThrow(git_repository_set_ident(data(), name.toLatin1(), email.toLatin1()));
```

Comment:
This function should have a way to clear the identity, just like `git_repository_set_ident` has (by providing `NULL`s as arguments). Perhaps, if `[name|email].isNull()`, pass `NULL` to `git_repository_set_ident`?

Reply:
The function has been modified and documented to be pretty much exactly like the corresponding function in libgit2 API.

Reply:
The `QString::isEmpty()` that is now used is indeed better than my proposed `isNull()`. Although it's not documented, deep inside libgit2 an error would be produced, if either name or email would be a zero-length string.